qbit-http-storages/src/commonMain/kotlin/qbit.storage/YandexDiskStorage.kt

```diff
@@ -0,0 +1,250 @@
+package qbit.storage
+
+import io.ktor.client.HttpClient
+import io.ktor.client.features.ClientRequestException
+import io.ktor.client.request.get
+import io.ktor.client.request.header
+import io.ktor.client.request.parameter
+import io.ktor.client.request.put
+import io.ktor.client.response.HttpResponse
+import io.ktor.client.response.readBytes
+import io.ktor.client.response.readText
+import io.ktor.content.ByteArrayContent
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.InternalCoroutinesApi
+import kotlinx.coroutines.launch
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonConfiguration
+import kotlinx.serialization.json.JsonObject
+import qbit.ns.Key
+import qbit.ns.Namespace
+import qbit.serialization.Storage
+
+class YandexDiskStorage(private val accessToken: String) : Storage {
+
+    private val YANDEX_DISK_API_GET_FILE_URL = "https://cloud-api.yandex.net/v1/disk/resources/upload"
+    private val YANDEX_DISK_API_GET_DOWNLOAD_FILE_URL = "https://cloud-api.yandex.net:443/v1/disk/resources/download"
+    private val YANDEX_DISK_API_RESOURCES = "https://cloud-api.yandex.net:443/v1/disk/resources"
+
+    @InternalCoroutinesApi
+    override fun add(key: Key, value: ByteArray) = runBlocking {
+        val client = HttpClient()
+        val paths = generatePutResourcePaths(key.ns)
+        try {
+            createFolderStructure(client, paths)
+        } catch (e: ClientRequestException) {
+            println(e)
+        }
+        val json = Json.indented.parseJson(getUrlToUploadFile(client, paths[paths.size - 1], key.name).readText())
+        var urlToUploadFile = (json as JsonObject).getPrimitive("href").content
+        var filePutResponse = uploadFile(client, urlToUploadFile, value)
+        client.close()
+    }
+
+    @InternalCoroutinesApi
+    override fun overwrite(key: Key, value: ByteArray) = runBlocking {
+        val client = HttpClient()
+        val paths = generatePutResourcePaths(key.ns)
+        val json = Json.indented.parseJson(getUrlToUploadFile(client, paths[paths.size - 1], key.name).readText())
```

Comment:
This code were just check is it possible to get data without deserailization, replace is with deseralization into data classes